ad_merge_strats.md

```diff
@@ -38,18 +38,43 @@ Adding the --no-ff will seperate the last few commits on a seperate branch so we
 git merge -s recursive feature_branch main_branch
 ```
 
-Recursive is the default merging strategy in Git when pulling or merging one branch. After branching and making some commits on the feature branch, there are typically some new commits on the main branch. Thus when branching, git recurses over the branch and creates a new merge commit, which will continue to have two parents. Git will also create a new ancestor commit if it cannot find any common ancestors.
+Recursive is the default merging strategy in Git when pulling or merging one branch and it uses three-way algorithm to merge two branches. After branching and making some commits on the feature branch, there are typically some new commits on the main branch. Thus when branching, git recurses over the branch and creates a new merge commit, which will continue to have two parents. Git will also create a new ancestor commit if it cannot find any common ancestors.
+
+<img src="images/recursive.png" alt="Visualization recursive" width="600"/>
+
+### Subtree
+
+```
+git merge -s subtree branch_A branch_B
+```
 
-<img src="recursive.png" alt="Visualization recursive" width="600"/>
+Subtree is an extension of the recursive strategy when branch B is a child subtree of branch A. The subtree strategy will first update the changes of the tree structure of branch A and then it will merge the changes just as the recursive strategy.
 
+The subtree strategy is also often used when pulling subrepository changes to parent repository.
 
 ### Resolve
 
+```
+git merge -s resolve feature_branch main_branch
+```
+
+Resolve strategy also uses three-way algorithm to merge two branches as does the recursive strategy. It looks for common commit ancestor of the two branches to base the merge on. Unlike recursive strategy, if it cannot find any common ancestor it will choose one randomly rather then creating a new one. 
```

Comment:
I suggest adding a comma into "...if it cannot find any common ancestor **","** it will choose...

Reply:
fixed